modules/apps/ccv/parent/keeper/relay.go

```diff
@@ -0,0 +1,80 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/ibc-go/modules/apps/ccv/parent/types"
+	ccv "github.com/cosmos/ibc-go/modules/apps/ccv/types"
+	clienttypes "github.com/cosmos/ibc-go/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/modules/core/24-host"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+func (k Keeper) SendPacket(ctx sdk.Context, chainID string, valUpdates []abci.ValidatorUpdate) error {
+	packetData := ccv.NewValidatorSetChangePacketData(valUpdates)
+	packetDataBytes := packetData.GetBytes()
+
+	channelID, ok := k.GetChainToChannel(ctx, chainID)
+	if !ok {
+		return sdkerrors.Wrapf(channeltypes.ErrChannelNotFound, "channel not found for channel ID: %s", channelID)
+	}
+	channel, ok := k.channelKeeper.GetChannel(ctx, types.PortID, channelID)
+	if !ok {
+		return sdkerrors.Wrapf(channeltypes.ErrChannelNotFound, "channel not found for channel ID: %s", channelID)
+	}
+	channelCap, ok := k.scopedKeeper.GetCapability(ctx, host.ChannelCapabilityPath(types.PortID, channelID))
+	if !ok {
+		return sdkerrors.Wrap(channeltypes.ErrChannelCapabilityNotFound, "module does not own channel capability")
+	}
+
+	// get the next sequence
+	sequence, found := k.channelKeeper.GetNextSequenceSend(ctx, types.PortID, channelID)
+	if !found {
+		return sdkerrors.Wrapf(
+			channeltypes.ErrSequenceSendNotFound,
+			"source port: %s, source channel: %s", types.PortID, channelID,
+		)
+	}
+
+	// Send ValidatorSet changes in IBC packet
+	packet := channeltypes.NewPacket(
+		packetDataBytes, sequence,
+		types.PortID, channelID,
+		channel.Counterparty.PortId, channel.Counterparty.ChannelId,
+		clienttypes.Height{}, uint64(types.GetTimeoutTimestamp(ctx.BlockTime()).UnixNano()),
+	)
```

Comment:
This can trigger OnTimeoutPacket. Could we have some ```None```  for the timestamp ? 
The current definition equals with the unbending time and does not consider any delays from the relayer (which comes at best empirical guarantees).  

For ordered channels we should close the channel due to an expired/ timeout valSetChange, which seems a bit harsh, since (at least for now) it's not a user value.

Reply:
Currently, a timeout must be set for now, but I think in a future version of core IBC, we may do this.

> For ordered channels we should close the channel due to an expired/ timeout valSetChange

This does seem very harsh. What is the safety risk in the case where we don't receive valset update by the unbonding period? There doesn't seem to be any at first glance, but I could be wrong